### PR TITLE
dependency: Updated Maven version in pom.xml

### PR DESCRIPTION
**Pull request by suniti0804**
Running maven with the current version 3.3.9 (as mentioned in pom.xml) results in the following error message - 

`[ERROR] Failed to execute goal org.codehaus.mojo:xml-maven-plugin:1.1.0:check-format (default) on project checkstyle: The plugin org.codehaus.mojo:xml-maven-plugin:1.1.0 requires Maven version 3.5.4 -> [Help 1]`

This PR updates the maven version from 3.3.9 to 3.5.4

### CODE REVIEW

**Comment by suniti0804** [comment]
@romani please add hacktoberfest-accepted tag

**Review by rnveach — CHANGES_REQUESTED** [review]
Missing documentation update.

> updates the maven version from 3.3.9 to 3.5.4

https://checkstyle.org/beginning_development.html
> Ensure that ... maven >= 3.3.9 are installed

What changed?

**Comment by rnveach** [comment]
> What changed?

`xml-maven-plugin` upgraded their maven version as part of 1.1.0 .
https://github.com/mojohaus/xml-maven-plugin/commit/9ae4e72844780e18cc49bf2a63df5f8fea94f857

We upgraded this version 4 months ago.
https://github.com/checkstyle/checkstyle/commit/4a69ab7991ea831ac6f75a4219648dc3c3b34a44

We should add a CI item to check maven version compatibility.

**Comment by suniti0804** [comment]
<groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-enforcer-plugin</artifactId>
        
this plugin requires maven version of 3.5.4

**Comment by rnveach** [comment]
Our version of `maven-enforcer-plugin` is 3.4.1, and it uses the same version of `enforcer` which specifies the maven version as `3.2.5`. So unless you are seeing this error, it may only be `xml-maven-plugin` that uses a newer version.

**Comment by suniti0804** [comment]
@rnveach can you please explain to me the cause of this error, this error led me to the conclusion that we might need an update of the above mentioned group and artifcact id, i think i am mistaken here - 
`[ERROR] Failed to execute goal org.codehaus.mojo:xml-maven-plugin:1.1.0:check-format (default) on project checkstyle: The plugin org.codehaus.mojo:xml-maven-plugin:1.1.0 requires Maven version 3.5.4 -> [Help 1]`

**Comment by rnveach** [comment]
My explanation of the cause of this error is at https://github.com/checkstyle/checkstyle/pull/13887#issuecomment-1763060926 .

Your post at https://github.com/checkstyle/checkstyle/pull/13887#issuecomment-1763063348 just mentions the wrong group/artifact.

What is left to be done is:
https://github.com/checkstyle/checkstyle/pull/13887#pullrequestreview-1678247981
> Missing documentation update.

And possibly:
https://github.com/checkstyle/checkstyle/pull/13887#issuecomment-1763060926
> We should add a CI item to check maven version compatibility.

**Comment by romani** [comment]
my local:
```
$ mvn --version
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /opt/maven/apache-maven-3.8.4
Java version: 11.0.14.1, vendor: Amazon.com Inc., runtime: /opt/jvm/amazon-corretto-11.0.14.10.1-linux-x64
Default locale: ru_RU, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-210-generic", arch: "amd64", family: "unix"
```

looks like we definitely need to update minimal version of maven.
Sad that non of CI jobs user a version that we declare in pom to report problem to us automatically

**Comment by romani** [comment]
@suniti0804 , please update this line also:
https://github.com/checkstyle/checkstyle/blob/b1b25109068ae4aec1d7042ceb9756bc02118c42/src/xdocs/beginning_development.xml#L20

**Comment by suniti0804** [comment]
@romani all the changes requested in https://github.com/checkstyle/checkstyle/pull/13887#issuecomment-1763415671 are implemented

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
Something weird is going on
https://app.travis-ci.com/github/checkstyle/checkstyle/builds/266600699#L198

**Comment by rnveach** [comment]
CI is failing.

> so lets update version to 3.8.6 not a 3.5.4.

I would stick to the lowest maven we can use. That is the point of this in the documentation.

Besides, CI is saying not all CIs have the latest maven installed.

**Comment by suniti0804** [comment]
@romani on it

File: pom.xml
Comment (romani) [inline pom.xml:1332]: to unblock this PR, please revert this line.
it is blocked by https://github.com/checkstyle/checkstyle/issues/13970 and it is heavy lifting operation as you can see in related PRs.

we can merge all other changes meanwhile.

File: src/xdocs/beginning_development.xml
Comment (romani) [inline src/xdocs/beginning_development.xml:20]: revert this change for now.

File: .travis.yml
Comment (romani) [inline .travis.yml:19]: move to `install:`

File: .travis.yml
Comment (romani) [inline .travis.yml:18]: remove this line we have it in "install:"

**Review by romani — CHANGES_REQUESTED** [review]
in this PR we just can update mvn versions in CI and merged PR.
actual update of min version will be done in scope of https://github.com/checkstyle/checkstyle/issues/13970
and updates in this PR will simplify a lot such future update.
 

items:

**Comment by romani** [comment]
After you addressed review item please reply "done" to explicitly show that you addressed this item in best way you understand it. I will review them and resolve if I agree.

CI are waiting please investigate and try to fix them.

File: azure-pipelines.yml
Comment (romani) [inline azure-pipelines.yml:137]: lets use https://stackoverflow.com/a/52343639

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: azure-pipelines.yml
Comment (romani) [reply azure-pipelines.yml:137]: No need to update azure, it has latest - 3.8.8
https://github.com/checkstyle/checkstyle/pull/14078#issuecomment-1832825993
We are ok to run mvnw that is above min required.

**Comment by nrmancuso** [comment]
@suniti0804 please resolve conflicts

**Pull request closed without merging**
